src/ynEIGEN/EigenStrategyManager.sol

```diff
@@ -0,0 +1,294 @@
+
+        ITokenStakingNode[] memory nodes = tokenStakingNodesManager.getAllNodes();
+        uint256 nodesCount = nodes.length;
+        uint256 assetsCount = assets.length;
+        for (uint256 j = 0; j < assetsCount; j++) {      
+
+            IERC20 asset = assets[j];
+            for (uint256 i; i < nodesCount; i++ ) {
+                ITokenStakingNode node = nodes[i];
+                uint256 balanceNode = asset.balanceOf(address(node));
+                stakedBalances[j] += balanceNode;
+
+                uint256 strategyBalance = toUserAssetAmount(
+                    asset,
+                    strategies[asset].userUnderlyingView((address(node)))
+                );
+                stakedBalances[j] += strategyBalance;
+            }
+        }
+    }
+
+    /**
+     * @notice Converts the user's underlying asset amount to the equivalent user asset amount.
+     * @dev This function handles the conversion for wrapped staked ETH (wstETH) and wrapped other ETH (woETH),
+     *      returning the equivalent amount in the respective wrapped token.
+     * @param asset The ERC20 token for which the conversion is being made.
+     * @param userUnderlyingView The amount of the underlying asset.
+     * @return The equivalent amount in the user asset denomination.
+     */
+    function toUserAssetAmount(IERC20 asset, uint256 userUnderlyingView) public view returns (uint256) {
+        if (address(asset) == address(wstETH)) {
+            // Adjust for wstETH using view method, converting stETH to wstETH
+            return wstETH.getWstETHByStETH(userUnderlyingView);
+        }
+        if (address(asset) == address(woETH)) { 
+            // Adjust for woETH using view method, converting oETH to woETH
+            return woETH.previewDeposit(userUnderlyingView);
+        }
+        return userUnderlyingView;
+    }
+
+    /**
+     * @notice Retrieves the total staked balance of a specific asset across all nodes.
+     * @param asset The ERC20 token for which the staked balance is to be retrieved.
+     * @return stakedBalance The total staked balance of the specified asset.
+     */
+    function getStakedAssetBalance(IERC20 asset) public view returns (uint256 stakedBalance) {
+        ITokenStakingNode[] memory nodes = tokenStakingNodesManager.getAllNodes();
+        uint256 nodesCount = nodes.length;
+        for (uint256 i; i < nodesCount; i++ ) {
+            ITokenStakingNode node = nodes[i];
+            stakedBalance += getStakedAssetBalanceForNode(asset, node);
+        }
+    }
+
+    /**
+     * @notice Retrieves the staked balance of a specific asset for a given node.
+     * @param asset The ERC20 token for which the staked balance is to be retrieved.
+     * @param node The specific node for which the staked balance is to be retrieved.
+     * @return stakedBalance The staked balance of the specified asset for the given node.
+     */
+    function getStakedAssetBalanceForNode(
+        IERC20 asset,
```

Comment:
missing validation that this actually an asset supported by this contract (`EigenStrategyManager`)

not an issue when called through `getStakedAssetBalance`, but when called directly you can pass in any `asset` (= `address`).